### PR TITLE
feat: Create Sidebar in chat page

### LINKED ISSUE #16: Create sidebar in chat page
- Add sidebar in chat page
- Use shadcn [sidebar](https://ui.shadcn.com/docs/components/sidebar)

### PR DESCRIPTION
**Pull request by mejsiejdev**
This PR adds the first component from the chat page, Sidebar.
Since the chat page has not been created yet, I made it under the link / directory `/chat`.

### CODE REVIEW

File: src/components/ChatSidebar/index.tsx
Comment (chewmanji) [inline src/components/ChatSidebar/index.tsx:1]: huh? Is it needed?

File: src/components/ChatSidebar/ChatSidebar.tsx
Comment (chewmanji) [inline src/components/ChatSidebar/ChatSidebar.tsx]: Titles of buttons are leftovers from previous implementation I guess

File: src/components/ChatSidebar/ChatSidebarLink.tsx
Comment (chewmanji) [inline src/components/ChatSidebar/ChatSidebarLink.tsx:13]: I'm not sure if we will need link there, I think that a button itself will be enough (to rerender chat) - but for now it's fine

File: src/components/ChatSidebar/index.tsx
Comment (mejsiejdev) [reply src/components/ChatSidebar/index.tsx:1]: no, I just wanted the import be cleaner

File: src/components/ChatSidebar/ChatSidebar.tsx
Comment (mejsiejdev) [reply src/components/ChatSidebar/ChatSidebar.tsx]: yup, forgot to change them 😭

File: src/components/ChatSidebar/ChatSidebarLink.tsx
Comment (mejsiejdev) [reply src/components/ChatSidebar/ChatSidebarLink.tsx:13]: how would that work though?

Comment (chewmanji) [reply src/components/ChatSidebar/ChatSidebarLink.tsx:13]: By passing function to button `onClick` method which will update the state in chat component, I guess.
Update function would take data from local storage and cause chat rerender.
But maybe I'm wrong, it would be good to hear @maciejkrol18

**Review by chewmanji — APPROVED** [review]

File: src/components/ChatSidebar/index.tsx
Comment (chewmanji) [reply src/components/ChatSidebar/index.tsx:1]: In shadcn components I noticed the convention of exporting stuff at the end of file:
`export {GreatLookingComponent, GreatLookingSubComponent, Dupa}`
That's may be more readable if you would have many compononents in one file but in our case (1 component in file) I don't think it's necessary
EDIT: nvm I see what you mean now - let's leave it as it is

File: src/components/ChatSidebar/ChatSidebarLink.tsx
Comment (mejsiejdev) [reply src/components/ChatSidebar/ChatSidebarLink.tsx:13]: I will wait for his review before I merge this

Comment (maciejkrol18) [reply src/components/ChatSidebar/ChatSidebarLink.tsx:13]: I think that how we handle this depends on how we're going to store user chats. If i recall correctly, currently we're going for a local storage approach, with all the chats being stored in user's browser, as to not have a need for implementing any auth and database. If we're going to store them in an array of objects, where each object represents a conversation with an UUID and rest of any relevant data (such as the messages), we could go for an approach resembling ChatGPT.

In ChatGPT, the users starts at an initial view with just the prompt bar ("Chat" view equivalent in our Figma design) with the route being `chatgpt.com`. Once the user submits their prompt, new conversation with an UUID gets created and the user gets redirect to a new route being `chatgpt.com/c/UUID` ("Chat with promochator" view equivalent   in our Figma design), which shows the conversation. 

How would this apply to our project? The "root" route would be `promochator.solvro.pl/chat` (equivalent of `chatgpt.com`). Each link in the sidebar would point to `promochator.solvro.pl/chat/UUID`. Upon entering this route, we would check if a conversation with matching UUID exists in user's browser storage and render it if found. "New chat" button would just point to `promochator.solvro.pl/chat`. 

If we would go for such approach then this piece of code is fine.

File: src/components/ui/sidebar.tsx
Comment (maciejkrol18) [inline src/components/ui/sidebar.tsx:294]: Does not match our Figma design, although there's no Lucide icon that looks like what we have there (two bars stacked on top of each other). Imo we might as well leave it as it is right now. 

@muclx is our UI/UX guy so I think this decision should be ultimately left to him

Comment (mejsiejdev) [reply src/components/ui/sidebar.tsx:294]: this icon is not part of the sidebar, it can be changed later

**Pull request merged**
